From 3a97acb56345c9dd5b1e1a6fb6559e0606e4df01 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Thu, 17 Nov 2011 00:46:18 +0800
Subject: [PATCH 243/442] ENGR00162344 vpu: Change unit test for interlaced
 stream per updated V4L

The new V4L interface needs to not dequeue last 2 frames on 2.6.38
kernel, so change the vpu unit test to align with in for mx6q.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 test/mxc_vpu_test/display.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/test/mxc_vpu_test/display.c b/test/mxc_vpu_test/display.c
index 6b14335..c587d21 100644
--- a/test/mxc_vpu_test/display.c
+++ b/test/mxc_vpu_test/display.c
@@ -889,8 +889,12 @@ void v4l_display_close(struct vpu_display *disp)
 		}
 
 		/* take care of ENGR00161948 */
-		if (cpu_is_mx6q())
-			threshold = 1;
+		if (cpu_is_mx6q()) {
+			if (disp->buf.field == V4L2_FIELD_ANY || disp->buf.field == V4L2_FIELD_NONE)
+				threshold = 1;
+			else
+				threshold = 2;
+		}
 
 		while (disp->queued_count > threshold) {
 			disp->buf.type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
-- 
1.8.0

